src/search.ts

```diff
@@ -92,7 +94,9 @@ export async function findFilesToUpload(
     directories so filter any directories out from the raw search results
   */
   for (const searchResult of rawSearchResults) {
-    if (!lstatSync(searchResult).isDirectory()) {
+    const fileStats = await stats(searchResult)
+    // isDirectory() returns false for symlinks if using fs.lstat(), make sure to use fs.stat() instead
+    if (!fileStats.isDirectory()) {
```

Comment:
Did you test with a symlink loop here?  What would happen?

Reply:
Loops would not be included as part of the `rawSearchResults`. [@actions/glob](https://github.com/actions/toolkit/blob/ccad19055eadcd85dd8bf73a487385c9c7233b4e/packages/glob/src/internal-globber.ts#L126) checks beforehand for broken symlinks and cycles and removes them.

We filter out directories after `@actions/glob` has finished searching